src/user-settings/user-settings.ts

```diff
@@ -0,0 +1,106 @@
+import {
+  BBAuth
+} from '../auth';
+
+import {
+  BBCsrfXhr
+} from '../shared/csrf-xhr';
+
+import {
+  BBUserConfig
+} from './user-config';
+
+const URL = 'https://sky-pusa01.app.blackbaud.net/uicfg/settings/user';
+
+let updateTimeoutId: any;
+
+export class BBUserSettings {
+
+  public static readonly UPDATE_DELAY = 1000;
+
+  public static readonly LOCAL_STORAGE_KEY = 'auth-client-local-user-settings';
+
+  public static getSettings(): Promise<BBUserConfig> {
+    return new Promise((resolve, reject) => {
+      BBAuth.getToken({
+        disableRedirect: true
+      }).then(
+        (token) => {
+          BBCsrfXhr.requestWithToken(
```

Comment:
Do we want a timeout here? We eventually put a timeout of 5 seconds in our library.

Reply:
Good catch. Done.